packages/sdk/src/dataProtector/sharing/setProtectedDataToRenting.ts

```diff
@@ -0,0 +1,65 @@
+import { WorkflowError } from '../../utils/errors.js';
+import {
+  isCollectionOwner,
+  isProtectedDataInCollection,
+} from '../../utils/sharing.js';
+import { throwIfMissing } from '../../utils/validators.js';
+import {
+  IExecConsumer,
+  SetProtectedDataToRentingParams,
+  SetProtectedDataToRentingResponse,
+  SubgraphConsumer,
+} from '../types.js';
+import { getSharingContract } from './smartContract/getSharingContract.js';
+
+export const setProtectedDataToRenting = async ({
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
+  collectionTokenId = throwIfMissing(),
+  protectedDataAddress = throwIfMissing(),
+  priceInNRLC = throwIfMissing(),
+  durationInSeconds = throwIfMissing(),
+}: IExecConsumer &
+  SubgraphConsumer &
+  SetProtectedDataToRentingParams): Promise<SetProtectedDataToRentingResponse> => {
+  //TODO:Input validation
+  const userAddress = await iexec.wallet.getAddress();
+  if (
+    !(await isCollectionOwner({
+      graphQLClient,
+      collectionTokenId: collectionTokenId,
+      walletAddress: userAddress,
+    }))
+  ) {
+    throw new WorkflowError(
+      'Failed to Set Protected Data To Renting: user is not collection owner.'
```

Comment:
This can be a misleading error message, it could also be that the collection hasn't been found.

Reply:
for that we need to add another check before this check if the collection is found or not.